module/zfs/zpl_ctldir.c

```diff
@@ -99,18 +99,35 @@ zpl_root_readdir(struct file *filp, void *dirent, filldir_t filldir)
  * Get root directory attributes.
  */
 /* ARGSUSED */
+
 static int
-zpl_root_getattr(struct vfsmount *mnt, struct dentry *dentry,
-    struct kstat *stat)
+zpl_root_getattr(
+#if defined(HAVE_3ARGS_IOPS_GETATTR)
+    struct vfsmount *mnt,
+    struct dentry *dentry,
+    struct kstat *stat
+#elif defined(HAVE_5ARGS_IOPS_GETATTR)
+    const struct path *path,
+    struct dentry *dentry,
+    struct kstat *stat,
+    u32 request_mask,
+    unsigned int query_flags
+#endif /* defined(HAVE_PATH_SIMPLE_GETATTR) */
+)
```

Comment:
You should use new form for this function, and make a wrapper function with old form inside ifdef